src/main/webapp/cluster-view/cluster-view.controller.js

```diff
@@ -1,26 +1,150 @@
 'use strict';
 
 angular.module('managementConsole')
-    .controller('ClusterViewCtrl', [
+  .controller('ClusterViewCtrl', [
     '$scope',
     '$stateParams',
     '$state',
+    '$q',
     'modelController',
-    function ($scope, $stateParams, $state, modelController) {
-            if (!modelController.isAuthenticated()) {
-                $state.go('/logout');
+    function ($scope, $stateParams, $state, $q, modelController) {
+      if (!modelController.isAuthenticated()) {
+        $state.go('/logout');
+      }
+      $scope.shared = {
+        currentCollection: 'caches'
+      };
+      $scope.clusters = modelController.getServer().getClusters();
+      $scope.currentCluster = modelController.getServer().getCluster($scope.clusters, $stateParams.clusterName);
+
+      $scope.$watch('currentCluster', function (currentCluster) {
+        if (currentCluster && currentCluster.name !== $stateParams.clusterName) {
+          $state.go('clusterView', {
+            'clusterName': currentCluster.name
+          });
+        }
+      });
+
+      $scope.currentClusterState = function () {
```

Comment:
This are really details, but maybe instead of implementing all this `currentCluster` functions, it may be nicer to implement this functions for any cluster (either as cluster methods or as function that takes cluster object) and then use them on current cluster, what do you think?